packages/tools/src/tools/displayTools/Labelmap/labelmapDisplay.ts

```diff
@@ -200,6 +204,27 @@ async function render(
   }
 
   for (const labelmapActorEntry of labelmapActorEntries) {
+    const uid = labelmapActorEntry.uid;
+    const labelmapActor = labelmapActorEntry.actor as vtkVolume | vtkImageSlice;
+    if (!labelmapActor || labelmapActor.isDeleted()) {
+      continue;
+    }
+
+    // Register onModified handler only once per actor UID
+    if (!onModifiedRegistered.has(uid)) {
```

Comment:
Hmm, what exactly constitutes a modified event on a VTK actor? I'm wondering if this impacts performance.

Reply:
You are right. I will put in the mapper so only when the data changes it will fire the event